mlflow/deployments/server/app.py

```diff
@@ -75,8 +75,13 @@ def get_dynamic_route(self, route_name: str) -> Optional[Route]:
 def _create_chat_endpoint(config: RouteConfig):
     prov = get_provider(config.model.provider)(config)
 
-    async def _chat(payload: chat.RequestPayload) -> chat.ResponsePayload:
-        return await prov.chat(payload)
+    async def _chat(
+        payload: chat.RequestPayload,
+    ) -> Union[chat.ResponsePayload, chat.StreamResponsePayload]:
+        if payload.stream:
+            return await make_streaming_response(prov.chat_stream(payload))
```

Comment:
if `prov.chat_stream()` returns a `HTTPException`, this can probably propagate it to the client